src/systems/systems/src/navigation/adirs.rs

```diff
@@ -1365,6 +1440,99 @@ impl InertialReference {
         self.update_wind_velocity(context, true_airspeed_source, overhead, simulator_data);
     }
 
+    fn update_maint_word(&mut self, overhead: &AirDataInertialReferenceSystemOverheadPanel) {
+        // TODO check status of these during mode transitions (first need to implement mode FSM)
+        let mut maint_word: IrMaintFlags = IrMaintFlags::default();
+
+        if !self.is_on {
+            // FIXME should be no transmission (can we just do this at a higher level...)
+            self.maint_word
+                .set_value(maint_word.bits(), SignStatus::FailureWarning);
+            return;
+        }
+
+        if !self.is_fully_aligned() {
+            maint_word |= IrMaintFlags::ALIGNMENT_NOT_READY;
+        }
+
+        if overhead.mode_of(self.number) == InertialReferenceMode::Attitude {
+            maint_word |= IrMaintFlags::REV_ATT_MODE;
+        }
+
+        if self.is_fully_aligned() {
+            maint_word |= IrMaintFlags::NAV_MODE;
+        }
+
+        // TODO request heading setting in att mode if not set
+
+        // TODO attitude invalid fault
+
+        // TODO dc < 18 V
+
+        // TODO on DC (re-implement ON BAT light properly at the same time)
+
+        // TODO ADR input data fault
+
+        // TODO unimportant nav fault
+
+        // TODO DC fault during DC operation last power up
+
+        // TODO align fault
+
+        // TODO No IRS initial pos
+
+        // TODO excess motion during align
+
+        // TODO ADR data not received or parity error
+
+        if self.extreme_latitude {
+            maint_word |= IrMaintFlags::EXTREME_LATITUDE;
+        }
+
+        if self.is_aligning() {
+            if self
```

Comment:
For some strange reason multiline comment no longer work. 

**suggestion**: You can replace this code with the following. Note that Rust 1.66 introduces half open ranges without lower bound. We're currently using 1.65. But later we can remove the lower bound on most of these matches and it would still work.

```rust
            maint_word |= match self
                .remaining_align_duration()
                .map(|duration| duration.as_secs())
            {
                Some(0..=60) => IrMaintFlags::ALIGN_1_MINUTES,
                Some(61..=120) => IrMaintFlags::ALIGN_2_MINUTES,
                Some(121..=180) => IrMaintFlags::ALIGN_3_MINUTES,
                Some(181..=240) => IrMaintFlags::ALIGN_4_MINUTES,
                Some(241..=300) => IrMaintFlags::ALIGN_5_MINUTES,
                Some(301..=360) => IrMaintFlags::ALIGN_6_MINUTES,
                Some(361..) => IrMaintFlags::ALIGN_7_10_MINUTES,
                None => IrMaintFlags::default(),
            };
```

Reply:
You could in fact also remove the `self.is_aligning()` check, and add a `Some(0)` condition with IrMaintFlags::default():

```rust
            maint_word |= match self
                .remaining_align_duration()
                .map(|duration| duration.as_secs())
            {
                Some(1..=60) => IrMaintFlags::ALIGN_1_MINUTES,
                Some(61..=120) => IrMaintFlags::ALIGN_2_MINUTES,
                Some(121..=180) => IrMaintFlags::ALIGN_3_MINUTES,
                Some(181..=240) => IrMaintFlags::ALIGN_4_MINUTES,
                Some(241..=300) => IrMaintFlags::ALIGN_5_MINUTES,
                Some(301..=360) => IrMaintFlags::ALIGN_6_MINUTES,
                Some(361..) => IrMaintFlags::ALIGN_7_10_MINUTES,
                Some(0) | None => IrMaintFlags::default(),
            };
```